src/pubsub/mod_pubsub.erl

```diff
@@ -308,48 +328,40 @@ init([ServerHost, Opts]) ->
     ets:insert(gen_mod:get_module_proc(ServerHost, config), {ignore_pep_from_offline, PepOffline}),
     ets:insert(gen_mod:get_module_proc(ServerHost, config), {host, Host}),
     ets:insert(gen_mod:get_module_proc(ServerHost, config), {access, Access}),
-    ets:insert(gen_mod:get_module_proc(ServerHost, config), {item_publisher, ItemPublisher}),
-    ejabberd_hooks:add(sm_remove_connection_hook, ServerHost,
-                       ?MODULE, on_user_offline, 75),
-    ejabberd_hooks:add(disco_local_identity, ServerHost,
-                       ?MODULE, disco_local_identity, 75),
-    ejabberd_hooks:add(disco_local_features, ServerHost,
-                       ?MODULE, disco_local_features, 75),
-    ejabberd_hooks:add(disco_local_items, ServerHost,
-                       ?MODULE, disco_local_items, 75),
-    ejabberd_hooks:add(presence_probe_hook, ServerHost,
-                       ?MODULE, presence_probe, 80),
-    ejabberd_hooks:add(roster_in_subscription, ServerHost,
-                       ?MODULE, in_subscription, 50),
-    ejabberd_hooks:add(roster_out_subscription, ServerHost,
-                       ?MODULE, out_subscription, 50),
-    ejabberd_hooks:add(remove_user, ServerHost,
-                       ?MODULE, remove_user, 50),
-    ejabberd_hooks:add(anonymous_purge_hook, ServerHost,
-                       ?MODULE, remove_user, 50),
-
-    case lists:member(?PEPNODE, Plugins) of
-        true ->
-            ejabberd_hooks:add(caps_recognised, ServerHost,
-                               ?MODULE, caps_recognised, 80),
-            ejabberd_hooks:add(disco_sm_identity, ServerHost,
-                               ?MODULE, disco_sm_identity, 75),
-            ejabberd_hooks:add(disco_sm_features, ServerHost,
-                               ?MODULE, disco_sm_features, 75),
-            ejabberd_hooks:add(disco_sm_items, ServerHost,
-                               ?MODULE, disco_sm_items, 75),
-            ejabberd_hooks:add(filter_local_packet, ServerHost, ?MODULE,
-                               handle_pep_authorization_response, 1),
-            gen_iq_handler:add_iq_handler(ejabberd_sm, ServerHost,
-                                          ?NS_PUBSUB, ?MODULE, iq_sm, IQDisc),
-            gen_iq_handler:add_iq_handler(ejabberd_sm, ServerHost,
-                                          ?NS_PUBSUB_OWNER, ?MODULE, iq_sm, IQDisc);
-        false ->
-            ok
-    end,
-    ejabberd_router:register_route(Host, mongoose_packet_handler:new(?MODULE, self())),
-    {_, State} = init_send_loop(ServerHost),
-    {ok, State}.
+    ets:insert(gen_mod:get_module_proc(ServerHost, config), {item_publisher, ItemPublisher}).
+
+hooks(Op, ServerHost) when Op == add; Op == delete ->
+    ejabberd_hooks:Op(sm_remove_connection_hook, ServerHost, ?MODULE, on_user_offline, 75),
```

Comment:
I'm not a fun of this solution with `Op`. Maybe it would not be that difficult to use `ejabberd_hooks:add/1` and `ejabberd_hooks:delete/1`?

Reply:
Sure :)